### PR TITLE
Support scm logger format

### PR DESCRIPTION
**Pull request by ngeiswei**
The logger scheme binding now supports scheme string formatting which allows to log things like:
```scheme
(cog-logger-info "My concept ~a" (ConceptNode "A"))
```
See http://wiki.opencog.org/wikihome/index.php/Scheme#Logger

### CODE REVIEW

File: opencog/scm/logger-utils.scm [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+(define (cog-logger-fine msg . args)
+  (cog-logger-fine-str (apply format #f msg args)))
+
+(define (export-logger-utils)
+  (export cog-logger-error
+          cog-logger-warn
+          cog-logger-info
+          cog-logger-debug
+          cog-logger-fine))
```
Comment (linas) [inline opencog/scm/logger-utils.scm]: The preferred method would be to move the entire contents of this file to `opencog/scm/opencog/logger.scm`

File: opencog/scm/rule-engine-utils.scm [hunk 2]
```diff
@@ -86,4 +86,8 @@
      rbs)
 )
 
-(define (export-rule-engine-utils) (export ure-add-rules ure-set-num-parameter ure-set-fuzzy-bool-parameter export-rule-engine-utils))
```
Comment (linas) [inline opencog/scm/rule-engine-utils.scm]: Likewise, all of this should go into `/opencog/scm/opencog/rule-engine.scm`

Comment (linas) [inline opencog/scm/rule-engine-utils.scm]: I'll make this change in a few mintes...

Comment (linas) [inline opencog/scm/rule-engine-utils.scm]: Pull request #383

File: opencog/scm/logger-utils.scm (hunk 1 above)
Comment (ngeiswei) [inline opencog/scm/logger-utils.scm]: I actually thought of that, but I just wasn't sure, can you express reasons why it should be the "preferred method"?

Comment (linas) [inline opencog/scm/logger-utils.scm]: No particularly strong reasons, other than its pointless to have too many itty bitty files.  If its a module, put it in the module. there is no plausible scenario where a user would try to load that file stand-alone without having previously loaded the module.

**Pull request merged**